src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -13,79 +11,96 @@ import { ApprovalState } from 'hooks/useApproveCallback'
 import { ButtonSize } from 'theme'
 import Loader from 'components/Loader'
 import { useErrorModal } from 'hooks/useErrorMessageAndModal'
+import { tryParseAmount } from 'state/swap/hooks'
+import { ONE_HUNDRED_PERCENT, ZERO_PERCENT } from 'constants/misc'
+import { PERCENTAGE_PRECISION } from 'constants/index'
 
-const RangeSteps = styled.div`
-  display: flex;
-  align-items: center;
-  width: 100%;
-  justify-content: space-between;
-`
-
-const RangeStep = styled.button`
-  background: none;
-  border: none;
-  font-size: 0.8rem;
-  cursor: pointer;
-  color: ${({ theme }) => theme.primary1};
-  padding: 0;
-`
-
-const INVESTMENT_STEPS = ['0', '25', '50', '75', '100']
-
-function _scaleValue(maxValue: CurrencyAmount<Currency>, value: string) {
-  // parse percent to string, example 25% -> 4 or 50% -> 2
-  const parsedValue = new Fraction(value, '100')
-
-  // divide maxValue with parsed value to get invest amount
-  return maxValue.multiply(parsedValue).asFraction
+enum ErrorMsgs {
+  Initial = 'Insufficient balance to cover input investment amount. Adjust the amount or go back to remove this investment option',
+  Balance = 'Insufficient balance to cover input investment amount, please adjust the input amount',
+  MaxCost = 'Input amount is bigger than available investment amount, please adjust the input amount',
 }
 
 export default function InvestOption({ approveData, claim, optionIndex }: InvestOptionProps) {
   const { currencyAmount, price, cost: maxCost } = claim
   const { updateInvestAmount } = useClaimDispatchers()
-  const { investFlowData } = useClaimState()
+  const { investFlowData, activeClaimAccount } = useClaimState()
 
   const { handleSetError, handleCloseError, ErrorModal } = useErrorModal()
 
-  const investedAmount = useMemo(() => investFlowData[optionIndex].investedAmount, [investFlowData, optionIndex])
+  const { account } = useActiveWeb3React()
 
-  const [percentage, setPercentage] = useState<string>(INVESTMENT_STEPS[0])
+  const [percentage, setPercentage] = useState<string>('0')
+  const [typedValue, setTypedValue] = useState<string>('0')
+  const [inputError, setInputError] = useState<string>('')
 
-  const { account } = useActiveWeb3React()
+  const investedAmount = investFlowData[optionIndex].investedAmount
 
   const token = currencyAmount?.currency
   const decimals = token?.decimals
   const balance = useCurrencyBalance(account || undefined, token)
 
-  const handleStepChange = useCallback(
+  const isSelfClaiming = account === activeClaimAccount
+  const noBalance = !balance || balance.equalTo('0')
+
+  // on invest max amount click handler
+  const setMaxAmount = useCallback(() => {
+    if (!maxCost || noBalance) {
+      return
+    }
+
+    const value = maxCost.greaterThan(balance) ? balance : maxCost
+    const amount = value.quotient.toString()
+
+    updateInvestAmount({ index: optionIndex, amount })
+    setTypedValue(formatSmart(value, decimals, { smallLimit: undefined }) || '')
```

Comment:
Are we using "formatSmart for the input field? That doesn't feel right. 

Format smart will for example write 12.56T for trillions. We want the numeric form in the input, right?

Reply:
I said the same thing :)